src/renderers/native/ReactNativeFiberErrorDialog.js

```diff
@@ -23,29 +23,25 @@ import type {CapturedError} from 'ReactFiberScheduler';
 function ReactNativeFiberErrorDialog(capturedError: CapturedError): boolean {
   const {componentStack, error} = capturedError;
 
-  let errorMessage: string;
-  let errorStack: string;
-  let errorType: Class<Error>;
+  let errorToHandle: Error;
 
   // Typically Errors are thrown but eg strings or null can be thrown as well.
-  if (error && typeof error === 'object') {
+  if (error instanceof Error) {
     const {message, name} = error;
 
     const summary = message ? `${name}: ${message}` : name;
 
-    errorMessage = `${summary}\n\nThis error is located at:${componentStack}`;
-    errorStack = error.stack;
-    errorType = error.constructor;
+    errorToHandle = error;
+    errorToHandle.message = `${summary}\n\nThis error is located at:${componentStack}`;
```

Comment:
If this is a frozen object (or non-writable) this will throw. Leaving the error unhandled.

We could use `Object.defineProperty(errorToHandle, 'message', { value: `${summary}\n\nThis error is located at:${componentStack}` })` but even that might throw for a non-configurable property.

Maybe we should just wrap this in try/catch?